grakn-graph/src/main/java/ai/grakn/graph/internal/concept/RelationshipImpl.java

```diff
@@ -95,11 +95,16 @@ public RelationshipStructure structure(){
 
     @Override
     public Relationship resource(Resource resource) {
-        reify().resource(resource);
+        resourceRelationship(resource);
         return this;
     }
 
     @Override
+    public Relationship resourceRelationship(Resource resource) {
```

Comment:
Here I'm defining `resource` in terms of `resourceRelationship` so I don't repeat myself.